### PR TITLE
Forgot Password flow

### LINKED ISSUE #101: Forgot My Password doesn't do anything

### LINKED ISSUE #112: Cannot login with mobile number
Try to login using a mobile number, and encounter this error:

image

### PR DESCRIPTION
**Pull request by rkdavidson**
# About
Closes #101
Closes #112

- Allows user to click _Forgot Password_ during login flow to send reset instructions via email
- Creates `/login/reset-password` page to mimic the one linked in that email
- Allows user to reset their password using email code

# Tradeoffs / Caveats
The page component for `ResetPassword` is a small monolith. It's a lot of lines, but it's ultimately a simple, standard form. We can decompose it if need be, but it felt like a premature optimization.

# Testing
- When logging in, click _Forgot Password_ to have an email sent with a code.
- The link in the email points to the current production site, but click it and copy/paste the `/login/reset-password?email={email}` portion to your local server URL
- Play with the reset flow. Leave required fields blank. Purposely make passwords mismatch. Etc.

# Screenshots & Gifs
|Caption|Gif|
|---|---|
|CleanShot_2021-04-27_23 53 59|Forgot Password|
|CleanShot_2021-04-27_23 55 13|Reset instructions email|
|CleanShot_2021-04-28_00 02 39|Success|
|errors|Common errors|
|reset form|Old Form|
|CleanShot_2021-04-28_00 15 11|New Form|

### CODE REVIEW

**Comment by asleepingpanda** [comment]
UI Looks incredible! I'm testing the flow now

File: providers/AuthProvider.js [hunk 1]
```diff
@@ -71,7 +71,12 @@ function AuthProvider(props = {}) {
       } else {
         if (!authenticated) {
           window.localStorage.removeItem(AUTH_TOKEN_KEY);
-          router.push('/');
+
```
Comment (jordanwade) [inline providers/AuthProvider.js:74]: @rkdavidson looks like there is a conflict with this and the other PR I just approved and merged. Do we need this here?

**Review by asleepingpanda — COMMENTED** [review]
## Password Reset Form
Confirmation Code should just be a numeric form field (all confirmation codes will be numeric, not alpha-numeric)

## Password Change is Successful
One of the following needs to happen:
* Log me in automatically
* Open up the Auth Modal with my email pre-filled

## I Click on the Button for "Forgot My Password"
What do you think about setting a 5 second timer that automatically redirects you to the `reset-password` screen with my email pre-filled? 
_Minor UX thing and not needed for launch. Just a thought_

File: providers/AuthProvider.js (hunk 1 above)
Comment (rkdavidson) [reply providers/AuthProvider.js:74]: Yeah was afraid of that, got my PRs mixed up a bit with the churn around this logout behavior.

I it's not needed, the router line and import/usage in this file can be removed. I can deal with this in a bit when I get home. Sorry about that.

**Comment by jordanwade** [comment]
@asleepingpanda  This is not a problem with this PR, but I noticed that the password reset form link in the email is not secure and should be changed to HTTPS. I got a warning the first time I clicked the link. Might want to add that to JIRA or something.

**Comment by jordanwade** [comment]
> @asleepingpanda This is not a problem with this PR, but I noticed that the password reset form link in the email is not secure and should be changed to HTTPS. I got a warning the first time I clicked the link. Might want to add that to JIRA or something.

Oops, just saw Ryans instructions in the PR description. NM

**Review by jordanwade — APPROVED** [review]

**Review by dzwood — DISMISSED** [review]
Everything is working awesome for me, nice @rkdavidson 🔥

File: providers/AuthProvider.js (hunk 1 above)
Comment (rkdavidson) [reply providers/AuthProvider.js]: Fixed in 35fe10d

**Pull request merged**